arch/sim/src/sim/posix/sim_alsa.c

```diff
@@ -581,6 +585,12 @@ static int sim_audio_ioctl(struct audio_lowerhalf_s *dev, int cmd,
 
           info->nbuffers    = priv->nbuffers;
           info->buffer_size = priv->buffer_size;
+
+          if (!priv->playback && priv->ops->get_samples)
```

Comment:
why check playback flag

Reply:
root cause is:  period time is 20ms,  just 3572 bytes in PCM.  But lame encoder need 1152*4 bytes once.  apb buffer not enough, so crash.  
playback flag  is not needed,  how about use MAX here?